src/matrix/lkj.jl

```diff
@@ -0,0 +1,219 @@
+"""
+struct LKJ{T <: Real, D <: Integer} <: ContinuousMatrixDistribution
+    d::D
+    η::T
+    logc0::T
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
+function LKJ(d::Integer, η::Real)
+    d > 0 || throw(ArgumentError("Matrix dimension must be positive."))
+    η > 0 || throw(ArgumentError("Scale parameter must be positive."))
+    logc0 = lkj_logc0(d, η)
+    T = Base.promote_eltype(η, logc0)
+    LKJ{T, typeof(d)}(d, T(η), T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+show(io::IO, d::LKJ) = show_multline(io, d, [(:d, d.d), (:η, d.η)])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{LKJ{T}}, d::LKJ) where T <: Real
+    LKJ{T, typeof(d.d)}(d.d, T(d.η), T(d.logc0))
+end
+
+function convert(::Type{LKJ{T}}, d::Integer, η, logc0) where T <: Real
+    LKJ{T, typeof(d)}(d, T(η), T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+dim(d::LKJ) = d.d
+
+size(d::LKJ) = (dim(d), dim(d))
+
+rank(d::LKJ) = dim(d)
+
+insupport(d::LKJ, R::AbstractMatrix) = isreal(R) && size(R) == size(d) && isone(Diagonal(R)) && isposdef(R)
+
+mean(d::LKJ) = Matrix{partype(d)}(I, dim(d), dim(d))
+
+mode(d::LKJ) = mean(d)
+
+var(d::LKJ) = (σ² = var(_marginal(d)); σ² * (ones(size(d)) - I))
+
+params(d::LKJ) = d.η
+
+@inline partype(d::LKJ{T}) where {T <: Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function lkj_logc0(d::Integer, η::Real)
+    if isone(η)
+        if iseven(d)
+            logc0 = lkj_onion_logc0_uniform_even(d)
+        else
+            logc0 = lkj_onion_logc0_uniform_odd(d)
+        end
+    else
+        logc0 = lkj_onion_logc0(d, η)
+    end
+    return logc0
+end
+
+logkernel(d::LKJ, R::AbstractMatrix) = (d.η - 1) * logdet(R)
+
+_logpdf(d::LKJ, R::AbstractMatrix) = logkernel(d, R) + d.logc0
+
+#  -----------------------------------------------------------------------------
+#  Sampling
+#  -----------------------------------------------------------------------------
+
+function _rand!(rng::AbstractRNG, d::LKJ, R::AbstractMatrix)
+    R .= _lkj_onion_sampler(d.d, d.η, rng)
+end
+
+function _lkj_onion_sampler(d::Integer, η::Real, rng::AbstractRNG = Random.GLOBAL_RNG)
+    #  Section 3.2 in LKJ (2009 JMA)
+    #  1. Initialization
+    R = ones(typeof(η), d, d)
+    β = η + 0.5d - 1
```

Comment:
Maybe 
```suggestion
    β = η + d/2 - 1
```
or should this have some `oftype`?

Reply:
β won't be promoted to whatever η is?

Reply:
```
η = 1f0
d = 4

julia> typeof(η), typeof(η + 0.5d - 1)
(Float32, Float64)
```

Reply:
Right, thanks. Since `Beta` suffers from #960 it doesn't end up mattering, unfortunately.

Reply:
Lets fix it anyway.